### PR TITLE
Program renaming

### PR DESCRIPTION
**Pull request by Molaryy**
# Description

I added the fronted for the program rename and also the backend, but I have a few bugs to resolve when the user needs to update the name.

# Changes include

- [ ] New feature (renaming file)

# Checklist

- [ ] I have tested with yarn build & yarn lint
- [ ] button rename & pencil icon
- [ ] car pop up that asks the program's name

- [ ] READMEs

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -1892,10 +1892,10 @@
     tweetnacl "^1.0.3"
     tweetnacl-util "^0.15.1"
 
-"@next/env@12.3.4":
-  version "12.3.4"
-  resolved "https://registry.yarnpkg.com/@next/env/-/env-12.3.4.tgz#c787837d36fcad75d72ff8df6b57482027d64a47"
-  integrity sha512-H/69Lc5Q02dq3o+dxxy5O/oNxFsZpdL6WREtOOtOM1B/weonIwDXkekr1KV5DPVPr12IHFPrMrcJQ6bgPMfn7A==
+"@next/env@13.3.4":
```
Comment (RezaRahemtola) [inline yarn.lock:1895]: why is the `yarn.lock` updated but not the `package.json` ? Looks like you switch to next v13, which could break some parts of the app

File: src/types/types.ts [hunk 2]
```diff
@@ -24,11 +24,14 @@ export type FileLog = {
 };
 
 export type IPCProgram = {
+	id: string;
 	hash: string;
+	encryptInfos: { key: string; iv: string };
```
Comment (RezaRahemtola) [inline src/types/types.ts:29]: I don't think you need this, you never encrypt the program in your code

File: src/types/types.ts [hunk 3]
```diff
@@ -24,11 +24,14 @@ export type FileLog = {
...
 	name: string;
 	createdAt: number;
 	entrypoint: string;
 	size: number;
+	logs: FileLog[];
```
Comment (RezaRahemtola) [inline src/types/types.ts:34]: If you start using it in programs you can rename simply rename it `Log`

File: src/lib/drive.ts [hunk 4]
```diff
@@ -7,11 +7,13 @@ import fileDownload from 'js-file-download';
 
 import { ALEPH_CHANNEL } from 'config/constants';
 
-import type { AggregateType, IPCContact, IPCFile, IPCFolder, ResponseType, UploadResponse } from 'types/types';
+import type { AggregateType, IPCContact, IPCFile, IPCFolder, ResponseType, UploadResponse , IPCProgram} from 'types/types';
 
 export const MONTH_MILLIS = 86400 * 30 * 1000;
 
 class Drive {
+	public programs: IPCProgram[];
```
Comment (RezaRahemtola) [inline src/lib/drive.ts:15]: The Drive contains logic related to files & folders, you shouldn't have programs stored in it :wink:

File: src/lib/contact.ts [hunk 5]
```diff
@@ -45,6 +46,47 @@ class Contact {
...
+		  );
+		  if (!programFound) {
+			const program = sharedPrograms.find((f) => f.id === concernedPrograms.id);
+			if (!program) {
+			  return { success: false, message: 'Program not found' };
+			}
+			await post.Publish({
+			  account: this.account,
+			  postType: 'InterPlanetaryCloud',
+			  content: { program: { ...concernedPrograms, name: newName }, tags: ['yolo', concernedPrograms.id] },
```
Comment (RezaRahemtola) [inline src/lib/contact.ts:78]: I think the key is `programs` (with an `s`). You added tags, is there a reason for this ? Do you understand what they are used for ?

File: src/components/programs/RenameProgram.tsx [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+	return (
+		<HStack
+			spacing={isDrawer ? '24px' : '12px'}
+			p="8px 12px"
+			borderRadius="8px"
+			role="group"
+			onClick={onOpen}
+			w="100%"
+			cursor="pointer"
+			id="ipc-dashboard-download-button"
```
Comment (RezaRahemtola) [inline src/components/programs/RenameProgram.tsx:84]: `ipc-dashboard-download-button` :thinking:

File: src/components/programs/RenameProgram.tsx [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+import { useUserContext } from 'contexts/user';
+import { useDriveContext } from 'contexts/drive';
+*/
+
+
+type RenameProgramProps = {
+	program: IPCProgram;
+};
+
+const RenameProgram =  ({ program}: RenameProgramProps): JSX.Element =>  {
```
Comment (RezaRahemtola) [inline src/components/programs/RenameProgram.tsx:36]: Looks like you still need to install Prettier correctly, it's not formatted here
```suggestion
const RenameProgram = ({ program }: RenameProgramProps): JSX.Element =>  {
```

File: src/lib/contact.ts (hunk 5 above)
Comment (Molaryy) [reply src/lib/contact.ts:78]: If I understood the tags are here to find the content easily right? if no can you please explain me this part.

**Pull request closed without merging**